### PR TITLE
tests: exercize disk plug/unplug instead of primary NIC

### PR DESCRIPTION
**Pull request by russell-islam**
Several live-migration, snapshot/restore, upgrade, and ivshmem tests hot-remove the primary virtio-net device and add it back mid-test to exercise PCI BAR reprogramming across a migration or restore. On the Ubuntu 22.04 guest this ometimes acily leaves the re-plugged interface in state DOWN (unmanaged): udev renames the freshly-plugged netdev (eth0 -> ensX) after systemd-networkd has already processed the initial RTM_NEWLINK, so networkd never re-runs its Match logic against the new name. The primary IP is never restored and the subsequent SSH check fails. This check makes several test cases flaky.

Drop the remove/add-back sequence from these tests. PCI BAR reprogramming on hotplug is already exercised by dedicated hotplug tests, and the migration/restore paths do not require it.

Instead, exercise hotplug during live migration/upgrade.

### CODE REVIEW

**Review by rbradford — CHANGES_REQUESTED** [review]
I thought you might replace it with a hot unplug/plug of a secondary block device?

**Comment by russell-islam** [comment]
With this change I don't see any flakiness on both MSHV and KVM.

**Comment by russell-islam** [comment]
> I thought you might replace it with a hot unplug/plug of a secondary block device?

Sorry, I misunderstood. Updated now.

**Comment by russell-islam** [comment]
One issue I found while I was exercising disk plug/unplug, If we hot plug a disk before upgrade and try to mount the disk as rw after upgrade mount fails. I am not sure if that use case is a valid test. opening for suggestions @weltling

File: cloud-hypervisor/tests/common/utils.rs [hunk 1]
```diff
@@ -1110,6 +1112,140 @@ pub(crate) fn bdf_from_hotplug_response(
...
+    /// After migration: re-mount the probe disk, verify the nonce, unmount,
+    /// then hot-remove the device on the destination.
+    pub(crate) fn post_migration_recheck(&self, guest: &Guest, dest_api_socket: &str) {
+        assert!(Self::exists(guest));
+
+        // Mount read-only: on the live-upgrade path the source is an older
+        // release that doesn't propagate `image_type=raw`, so the destination
+        // re-auto-detects the disk as raw and re-arms the sector-0 write
+        // guard. A read-write mount would then fail trying to update the
+        // ext4 superblock.
```
Comment (russell-islam) [inline cloud-hypervisor/tests/common/utils.rs:1222]: @weltling Here.

File: cloud-hypervisor/tests/common/utils.rs [hunk 2]
```diff
@@ -1110,6 +1110,73 @@ pub(crate) fn bdf_from_hotplug_response(
...
+/// Uses the 16 MiB `blk.img` workload disk that is already fetched for the
+/// disk-hotplug tests; when hot-added it appears in the guest as `/dev/vdc`.
+pub(crate) struct HotplugDiskProbe {
+    disk_id: &'static str,
+    params: String,
+}
+
+#[cfg(target_arch = "x86_64")]
+impl HotplugDiskProbe {
+    pub(crate) fn new() -> Self {
```
Comment (rbradford) [inline cloud-hypervisor/tests/common/utils.rs:1137]: No thanks.

**Review by rbradford — CHANGES_REQUESTED** [review]

File: cloud-hypervisor/tests/common/utils.rs (hunk 2 above)
Comment (rbradford) [reply cloud-hypervisor/tests/common/utils.rs:1137]: This is clearly AI slop without an `Assisted-by` trailer. Please be respectful of reviewers time by stopping and thinking and looking at the context and reviewing what the tool has produced. When used smartly LLMs are invaluable but if you don't actually invest time into considering the output, which ultimately has *your* name on it not only do you injure your reputation and you also negatively impact the project's codebase.

Comment (russell-islam) [reply cloud-hypervisor/tests/common/utils.rs:1137]: My agent some times does append the assisted by automatically when I try to fix typo of my commit meesage some times doesn't. THe reason I introduced this code becasue you asked me to exercise the disk hotplug and unplug instead of primary NIC. Adding same code to all test cases makes a lots of duplicate code. Th ewhole thing was my idea and I just used AI to assist me. Again I should have mentioned it. Let me know how can the new check be added with minimum duplicacy.

Comment (rbradford) [reply cloud-hypervisor/tests/common/utils.rs:1137]: My point is that we don't have a pattern with a helper like this in the code. It would be better to just to follow the existing style.

**Comment by russell-islam** [comment]
I think it's better to add this fixes to my other live-migration PR. To make the tests easier for me. And these two PRs are related. 
Closing this for now.
